client/src/observatory/utils.tsx

```diff
@@ -0,0 +1,90 @@
+export function Link({ href, children }: { href: string; children: any }) {
+  return (
+    <a
+      href={href}
+      target="_blank"
+      rel="noreferrer"
+      className={`${href.startsWith("/") ? "" : "external"}`}
+    >
+      {children}
+    </a>
+  );
+}
+
+export function PassIcon({ pass }: { pass: boolean | null }) {
+  if (pass === null) {
+    return <>-</>;
+  }
+  return (
+    <span className="obs-pass-icon">
+      {pass ? (
+        <svg
+          aria-hidden="true"
+          className="pass"
+          width="24"
+          height="27"
+          viewBox="0 -2 24 25"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <circle cx="12" cy="11.8984" r="9" />
+          <path d="M12 1.89844C6.5 1.89844 2 6.39844 2 11.8984C2 17.3984 6.5 21.8984 12 21.8984C17.5 21.8984 22 17.3984 22 11.8984C22 6.39844 17.5 1.89844 12 1.89844ZM10 16.8984L5 11.8984L6.41 10.4884L10 14.0684L17.59 6.47844L19 7.89844L10 16.8984Z" />
+        </svg>
```

Comment:
Why are the pass/fail SVGs inlined here?

Reply:
Moved inline svg data out into assets.